### ISSUE TITLE
Xml Cache Corruption Bug Report And Fix

### ISSUE THREAD

**Issue by metgithub** [umbraco/Umbraco-CMS#8965]
The Umbraco 7.15.6 preview gives error "The Xml cache is corrupt. Use the Health Check data integrity dashboard to fix it." even tho the cache is correct. Here is the bug and how i fixed it

## Umbraco version

I am seeing this issue on Umbraco version: 7.15.6 (Umbraco.Core 7.15.6)

Reproduction
------------

Install Umbraco 7.15.6 and click preview

### Bug summary

When you click the preview button, an ysod pop up stating "The Xml cache is corrupt. Use the Health Check data integrity dashboard to fix it"

### Specifics

What version of Umbraco are you using (down to the very last digit!): 7.15.6
What browser and version you are using: Chrome 85.0.4183.121
Please mention if you've checked it in other browsers as well: Yes
Please include *full error messages* and *screenshots* if possible: 
 * The Xml cache is corrupt. Use the Health Check data integrity dashboard to fix it

### Steps to reproduce

Click Preview

### Expected result

Seeing the preview 

### Actual result

Ysod: The Xml cache is corrupt. Use the Health Check data integrity dashboard to fix it

### HOW TO I FIXED IT:

The bug is in the Umbraco.Core DLL, in the Persistance Namespace inside the Database Class in the method BuildSqlDbSpecificPagingQuery:536
Here there is this line:
`int num = sqlSelectRemoved.IndexOf("FROM");`
But `sqlSelectRemoved` is in lowercase, and so IndexOf returns -1 and the next line is a Substring(0, -1) which throws an ArgumentException.

The fix actually consist in changing it with (I see you prefer type it in uppercase so if the string is already in uppercase there are no memory leaks):
`int num = sqlSelectRemoved.ToUpper().IndexOf("FROM");`

And then it starts working again.
Hope this will help 👍

**Comment by umbrabot** [comment]
Hi @metgithub,

We're writing to let you know that we would love some help with this issue. We feel that this issue is ideal to flag for a community member to work on it. Once flagged here, folk looking for issues to work on will know to look at yours. Of course, please feel free work on this yourself ;-).  If there are any changes to this status, we'll be sure to let you know. 

For more information about issues and states, have a look at [this blog post](https://umbraco.com/blog/the-umbraco-issue-tracker-process/)

Thanks muchly, from your friendly Umbraco GitHub bot :-)

**Comment by elit0451** [comment]
Seems like this is no longer a problem, so I will go ahead and close this 🙂 Apologies, for the late reply

**Comment by matusqo** [comment]
Hi all,

today I had the same issue in UAT environment even it was working in DEV environment. Finally it occurred to me that it could be the folder permissions as I deployed the website using CI/CD pipeline by copying all files from build machine. **I checked permissions of the root website foder that IIS application is mapped to, compared with DEV one and noticed that there is missing Application Pool identity in the list in UAT.** I added it and forced to update on all subfolders and files and it worked right after that!

**Based on that I now believe this error message is absolutely unrelated to any XML corruption or cache problem. This is a general error message that can be caused by dozens of reasons!**

![image](https://user-images.githubusercontent.com/50449453/228507813-7100aac7-e0d8-4c3b-9176-50e0ae2c4013.png)
